src/App.js

```diff
@@ -46,7 +47,38 @@ body {
 }
 `;
 
-const App = () => {
+const initializeState = ({ set }) => {
+  const keys = Object.keys(localStorage).filter(
+    (key) => !key.includes("mapbox")
+  );
+
+  const promises = keys.map((key) => localStorage.getItem(key));
+  Promise.all(promises).then((values) => {
+    for (let i = 0; i < promises.length; i++) {
+      const key = keys[i];
+      const value = JSON.parse(values[i]).value;
+      set({ key }, value);
+    }
+  });
+};
+
+const Inner = () => {
+  useTransactionObservation_UNSTABLE(
+    ({
+      atomValues,
+      previousAtomValues,
+      atomInfo,
+      modifiedAtoms,
+      transactionMetadata,
+    }) => {
+      for (const modifiedAtom of modifiedAtoms) {
+        localStorage.setItem(
+          modifiedAtom,
+          JSON.stringify({ value: atomValues.get(modifiedAtom) })
```

Comment:
unable to save data into local storage for now. recoil still buggy! atomValues values are not correctly set!